### PR TITLE
Proposal for Linux Seccomp addition to Marathon

### PR DESCRIPTION
**Pull request by kensipe**
Summary:
We need to add seccomp to Marathon.  This requires a Mesos bump to version 1.8.

JIRA issues: DCOS-50606

### CODE REVIEW

File: docs/docs/rest-api/public/api/v2/types/linux.raml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#%RAML 1.0 Library
+uses:
+  strings: stringTypes.raml
+
+types:
+  SecComp:
+    type: object
+    properties:
+      profile_name?:
```
Comment (timcharper) [inline docs/docs/rest-api/public/api/v2/types/linux.raml]:
```suggestion
      profileName?:
```

Camel case, as seen elsewhere.

File: docs/docs/rest-api/public/api/v2/types/linux.raml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        type: string
+        description: filename of the Seccomp profile found on the agent
+      unconfined?:
+        type: boolean
+        description:  set to `true` if Seccomp is not applied to the container
+  LinuxRuntime:
+    type: object
+    properties:
+      seccomp?: SecComp
```
Comment (timcharper) [inline docs/docs/rest-api/public/api/v2/types/linux.raml]: Any chance we can add this newline? super nitty and fine if not

File: docs/docs/rest-api/public/api/v2/types/podContainer.raml [hunk 3]
```diff
@@ -84,6 +85,7 @@ types:
         type: network.Endpoint[]
         (pragma.omitEmpty):
         description: The ports needed to run this container
+      linuxInfo?: linux.LinuxRuntime
```
Comment (meichstedt) [inline docs/docs/rest-api/public/api/v2/types/podContainer.raml]: would it make sense to rename the type so we have
```
linuxInfo?: linux.LinuxInfo
```
?

File: docs/docs/rest-api/public/api/v2/types/linux.raml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+uses:
+  strings: stringTypes.raml
+
+types:
+  Seccomp:
+    type: object
+    properties:
+      profileName?:
+        type: string
+        description: filename of the Seccomp profile found on the agent
```
Comment (meichstedt) [inline docs/docs/rest-api/public/api/v2/types/linux.raml]: Add a note like
> the task will fail if that file does not exist

?

File: docs/docs/rest-api/public/api/v2/types/linux.raml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      unconfined?:
+        type: boolean
+        description:  set to `true` if Seccomp is not applied to the container
```
Comment (meichstedt) [inline docs/docs/rest-api/public/api/v2/types/linux.raml]: > set to `true` if Seccomp shall not be applied to the container

?

File: docs/docs/rest-api/public/api/v2/types/appContainer.raml [hunk 6]
```diff
@@ -185,6 +186,7 @@ types:
         default: DOCKER
       docker?: DockerContainer
       appc?: AppCContainer
+      linuxInfo?: linux.LinuxRuntime
```
Comment (meichstedt) [inline docs/docs/rest-api/public/api/v2/types/appContainer.raml]: WDYT about
```
linuxInfo?: linux.LinuxInfo
```
?

File: docs/docs/rest-api/public/api/v2/types/podContainer.raml [hunk 7]
```diff
@@ -97,7 +99,7 @@ types:
           Values here override a "user" value specified in the pod definition.
       healthCheck?:
         type: health.HealthCheck
-        description: All healthchecks to perform on the container
+        description: All health checks to perform on the container
```
Comment (meichstedt) [inline docs/docs/rest-api/public/api/v2/types/podContainer.raml]: "The health check to perform on the container, if any." – there can be at most one.

Comment (kensipe) [reply docs/docs/rest-api/public/api/v2/types/podContainer.raml]: that will teach me to fix misspellings :)

**Comment by kensipe** [comment]
thanks for the feedback.   I didn't care for the "Info" of linuxInfo... Info is such a vague term as to not add value IMO.     I think "runtime" is closer.. but it makes sense to be consistent... and it is easy to claim traceability to the mesos defined "Info" of LinuxInfo.   Seems like we are there... thanks!

**Pull request merged**